src/pages/index.js

```diff
@@ -0,0 +1,283 @@
+
+  if (data.isLiked) {
+    cardLikeBtn.classList.add("card__like-button_liked");
+  }
+
+  cardNameEl.textContent = data.name;
+  cardImageEl.src = data.link;
+  cardImageEl.alt = data.name;
+
+  cardDeleteButton.addEventListener("click", () =>
+    handleDeleteCard(cardElement, data._id)
+  );
+  cardLikeBtn.addEventListener("click", (evt) =>
+    handleLikeButton(evt, data._id)
+  );
+  cardImageEl.addEventListener("click", () => {
+    openModal(previewModal);
+    previewImageEl.src = data.link;
+    previewImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+  });
+
+  return cardElement;
+}
+
+// Open & Close Modal Functions
+function handleLikeButton(evt, _id) {
+  const likeButton = evt.target;
+  const isLiked = likeButton.classList.contains("card__like-button_liked");
+
+  api
+    .handleLike(_id, isLiked)
+    .then(() => {
+      likeButton.classList.toggle("card__like-button_liked");
+    })
+    .catch(console.error);
+}
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+}
+
+function handleEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) closeModal(openedModal);
+  }
+}
+
+// DELETE card
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+
+  setButtonText(submitBtn, true, "Yes", "Deleting...");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(deleteModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Delete", "Deleting...");
+    });
+}
+
+// PROFILE edit
+function handleProfileFormEdit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .editUserInfo({
+      name: editModalName.value,
+      about: editModalDescription.value,
+    })
+    .then((data) => {
+      profileNameElement.textContent = data.name;
+      profileDescriptionElement.textContent = data.about;
+      closeModal(profileEditModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false, "Save", "Saving...");
+    });
+}
+
+// AVATAR update
+function handleAvatarSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+
+  setButtonText(submitBtn, true, "Save", "Saving...");
+
+  api
+    .editAvatarInfo(avatarInput.value)
+    .then((data) => {
+      profileAvatarElement.src = data.avatar;
+      closeModal(avatarModal);
+      avatarForm.reset();
```

Comment:
* Please, disable the `avatar form` and `card form` submit buttons after submission


The `disableButton` function requires 2 arguments: `buttonEl, config`. Please, pass the submit button and `settings` here


```js
      disableButton(submitBtn, settings);

```